src/openrct2/ride/Vehicle.cpp

```diff
@@ -8891,7 +8891,7 @@ int32_t Vehicle::UpdateTrackMotion(int32_t* outStation)
     UpdateVelocity();
 
     Vehicle* vehicle = this;
-    if (_vehicleVelocityF64E08 < 0)
+    if (_vehicleVelocityF64E08 < 0 && !vehicle->HasUpdateFlag(VEHICLE_UPDATE_FLAG_SINGLE_CAR_POSITION))
```

Comment:
I can't remember if we allowed this flag on mini golf. If we did we need to modify the same bit of mini golf code

Reply:
We did yes, thank you for the reminder! I've taken a quick look but the mini golf looks a lot more complicated and undocumented than the regular vehicle code. Any idea where the flag should be checked as well? I'll try to dive into it tomorrow as well.

 For reference, the update flag was added in #13607 and the plugin api to use it was added in #13593

Reply:
Update: the minigolf does not seem to have the bug I originally intended to fix. (Probably because there's no gravity involved with its vehicles?) 

It does have another bug though where backwards movement is quite glitchy.

https://user-images.githubusercontent.com/20048660/173187053-5c42f69e-424e-408e-9246-b91cd4be917c.mp4

This seems like another unrelated bug.

Reply:
Okay i investigated mini golf and it as you said has simplified velocity calcs which don't care for gravity